app/response_processor.py

```diff
@@ -106,33 +106,41 @@ def send_receipt(self, decrypted_json):
             receipt_json = {
                 'tx_id': decrypted_json['tx_id'],
                 'collection': {
-                    'exercise_sid': decrypted_json['collection']['exercise_sid']
+                    'exercise_sid':
+                    decrypted_json['collection']['exercise_sid']
                 },
                 'metadata': {
                     'ru_ref': decrypted_json['metadata']['ru_ref'],
                     'user_id': decrypted_json['metadata']['user_id']
                 }
             }
         except KeyError as e:
-            self.logger.error("Unsuccesful publish, missing key values", error=e)
+            self.logger.error(
+                "Unsuccesful publish, missing key values", error=e)
             raise QuarantinableError
 
+        try:
+            case_id = decrypted_json['case_id']
+            receipt_json['case_id'] = case_id
+        except KeyError:
+            # Don't do anything, as this survey originated from RRM
+            pass
+
         if not decrypted_json.get("survey_id"):
             self.logger.error("No survey id")
             raise QuarantinableError
-
         elif decrypted_json.get("survey_id") == "census":
             self.logger.info("Ignoring received CTP submission")
             return None
-
         else:
 
             try:
                 self.logger.info("About to publish receipt into rrm queue")
                 self.logger.debug(receipt_json)
-                self.rrm_publisher.publish(dumps(receipt_json),
-                                           headers={'tx_id': decrypted_json['tx_id']},
-                                           secret=settings.SDX_COLLECT_SECRET)
+                self.rrm_publisher.publish(
+                    dumps(receipt_json),
+                    headers={'tx_id': decrypted_json['tx_id']},
+                    secret=settings.SDX_COLLECT_SECRET)
             except PublishMessageError as e:
                 self.logger.error("Unsuccesful publish", error=e)
```

Comment:
There are a couple of these logger errors here, should we change to logger.exception?

Reply:
happy to raise it as tech debt - would need testing with Splunk etc which is outside scope of this PR